sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -105,41 +175,36 @@ public static bool IsNotIssueLocationLine(string line)
 
         internal /*for testing*/ IEnumerable<IssueLocation> GetIssueLocations(TextLine line)
         {
-            return GetLocations(line, ISSUE_LOCATION_PATTERN);
+            return GetLocations(line, RxIssue);
         }
 
         internal /*for testing*/ IEnumerable<IssueLocation> GetBuildErrorsLocations(TextLine line)
         {
-            return GetLocations(line, BUILD_ERROR_LOCATION_PATTERN);
+            return GetLocations(line, RxBuildError);
         }
 
-        private IEnumerable<IssueLocation> GetLocations(TextLine line, string pattern)
+        internal /*for testing*/ IEnumerable<IssueLocation> GetPreciseIssueLocations(TextLine line)
         {
             var lineText = line.ToString();
 
-            var match = Regex.Match(lineText, pattern);
+            var match = RxPreciseLocation.Match(lineText);
             if (match.Success)
             {
-                return CreateIssueLocations(match, line.LineNumber + 1);
+                EnsureNoRemainingCurlyBrace(line, match);
+                return CreateIssueLocations(match, line.LineNumber);
             }
-
             return Enumerable.Empty<IssueLocation>();
         }
 
-        internal /*for testing*/ IEnumerable<IssueLocation> GetPreciseIssueLocations(TextLine line)
+        private IEnumerable<IssueLocation> GetLocations(TextLine line, Regex rx)
         {
             var lineText = line.ToString();
 
-            var match = Regex.Match(lineText, PRECISE_ISSUE_LOCATION_PATTERN);
+            var match = rx.Match(lineText);
```

Comment:
Is the lading empty line needed? `lineText` is used only once, I would remove the variable and put `.ToString()` directly into `.Match(...)`.